### ISSUE TITLE
license tweak?

### ISSUE THREAD

**Issue by proppy** [nothings/stb#240]
The following license clause prevent me to re-use `stb_truetype.h` for [Android NDK samples](https://github.com/googlesamples/android-ndk).

> Where that dedication is not recognized

I was wondering if you would consider dropping the "Where that ..." condition part from the license grant or dual licensing the stb libraries under [Apache 2.0](http://www.apache.org/licenses/LICENSE-2.0)?

Thanks in advance.

**Comment by nothings** [comment]
Dropping that condition would make its legal meaning unclear where the public-domain dedication is unrecognized. Can you explain more clearly why the use of the clause prevents you from using stb_truetype, or what eliminating the clause accomplishes? (I can see the intent, but I don't know what the scenario is where it matters.)

**Comment by proppy** [comment]
@nothings ANAL, but the reason I was given was the "the license grant being contingent on a poorly defined condition" (condition == "Where that dedication is not recognized").

I can ask for more details if you're interested.

**Comment by nothings** [comment]
Please do. The problem with the recommendation is that I need the license to be acceptable in all jurisdictions, and I don't know if the lawyer(?) who raised this issue was considering all jurisdictions, or only their own.

**Comment by proppy** [comment]
@nothings will do, what do you think about also providing the library under a [popular opensource license](https://opensource.org/licenses).

**Comment by nothings** [comment]
It's in the public domain. Anyone who wants to can put any arbitrary license on it, creating a derivative work that is under that license. There's no reason for me to make one 'official'.

**Comment by proppy** [comment]
@nothings sadly, I'm afraid I don't have much more details to share: some company policy (including mine) is to only allow [OSI-approved licenses](https://opensource.org/licenses/alphabetical).

There is some detail about the complicated situation with ["public domain"](https://opensource.org/faq#public-domain) on their website along with some recommendations.

I'll close this issue, and *thank you* again for taking the time to answer my questions.

**Comment by jarikomppa** [comment]
It's kinda odd that the list of licenses includes a bunch of rather obscure
ones but doesn't include unlicense (for instance - I do understand why
they're not listing WTFPL). From a quick glance, the one closest to "public
domain" was this:

https://opensource.org/licenses/FPL-1.0.0

which is basically "use however you want as long as you don't sue me for it"

Another which is pretty close is

https://opensource.org/licenses/Fair

..that said, since the works are in public domain, what stops anyone from
applying any license on top of them (or their trivial changes) ?

The primary reason for putting things in public domain is to avoid this
discussion in the first place..

**Comment by nothings** [comment]
unlicense is just public domain so it makes sense it's not listed separately.

Note that the scenario in this thread is basically the same one I've run into before (see last two paragraphs of https://github.com/nothings/stb/blob/master/docs/why_public_domain.md). And the whole 'complexity described by OSI' is exactly the reason for the 'where that dedication is not recognized' section of the license which makes it pretty frustrating, but in the end my conclusion is: other people's lawyers are not my problem.

**Comment by proppy** [comment]
> Note that the scenario in this thread is basically the same one I've run into before 

@nothings, sorry for the noise, I should have look at the `why_public_domain.md` doc before filing a new issue.

**Comment by nothings** [comment]
No worries, it's nice to have one of these preserved publically in the issues.
